documentation/specs/proposed/BuildCheck-Architecture.md

```diff
@@ -0,0 +1,94 @@
+
+# BuildCheck - Architecture and Implementation Spec
+
+This is an internal engineering document. For general overview and user point of view - please check the [BuildCheck - Design Spec](BuildCheck.md).
+
+# Areas of Ownership
+
+| Area     |      Owner     |
+|----------|:-------------|
+| PM                  | @baronfel |
+| Advisory/Leadership | @rainersigwald |
+| Infrastructure      | @jankrivanek |
+| Configuration       | @f-alizada   |
+| Custom Analyzers    | @YuliiaKovalova |
+| Inbox Analyzers     | @ladipro |
+| Replay Mode         | @surayya-MS |
+| Tracing             | @maridematte |
+| Perf Advisory       | @AR-May |
+
+# Table of Contents
+
+- [Infrastructure and Execution](#infrastructure-and-execution)
+   * [Data Source](#data-source)
+   * [Execution Modes](#execution-modes)
+   * [Live Mode Hosting](#live-mode-hosting)
+   * [Handling the Distributed Model](#handling-the-distributed-model)
+   * [Analyzers Lifecycle](#analyzers-lifecycle)
+- [Configuration](#configuration)
+- [Acquisition](#acquisition)
+- [Build OM for Analyzers Authoring](#build-om-for-analyzers-authoring)
+
+# Infrastructure and Execution
+
+## Data Source
```

Comment:
I'd like to be clear here that we're going to have a mapping OM from the source to the things that are exposed to plugins.